### ISSUE TITLE
Request status 400 after upgrade from 0.10.1 to 0.10.2

### ISSUE THREAD

**Issue by uteminhduc** [pocketbase/pocketbase#1365]
I have issue after upgrade 0.10.2

My request:
`
http://localhost:8090/api/collections/tests/records/ws2q59uc581p7oa?expand=questions.choices%2Cquestions.childs.choices%2Ctags
`

_Childs field in the collection is related to itself_

**Respone at v0.10.2 and v0.10.3:**
_Request logs on Admin page did not record_
```json
{
  "code": 400,
  "message": "Something went wrong while processing your request.",
  "data": {}
}
```

**Respone at v0.10.1:**
```json
{
  "childs": [],
  "collectionId": "yg2xylk9a3jv7lb",
  "collectionName": "tests",
  "count": 6,
  "created": "2022-12-15 06:13:44.096Z",
  "description": "description",
  "expand": {
    "questions": [
      {
        "audio": "t1_uCroYDyvJ4.mp3",
        "childs": [],
        "choices": [
          "f7eve1wbozruzu6",
          "8wagn9mbhryp1sa",
          "3z270afy8bmgvcv",
          "lbalsgmn92sigp8"
        ],
        "collectionId": "v6i1h1klj8t22ef",
        "collectionName": "questions",
        "correct": "lbalsgmn92sigp8",
        "created": "2022-12-15 06:13:50.384Z",
        "expand": {
          "choices": [
            {
              "collectionId": "63e9guamova874l",
              "collectionName": "choices",
              "created": "2022-12-15 06:13:49.796Z",
              "explain": "",
              "id": "f7eve1wbozruzu6",
              "name": "",
              "transcript": "A box is being loaded onto a truck.",
              "updated": "2022-12-15 06:13:49.796Z"
            },
            {
              "collectionId": "63e9guamova874l",
              "collectionName": "choices",
              "created": "2022-12-15 06:13:49.912Z",
              "explain": "",
              "id": "8wagn9mbhryp1sa",
              "name": "",
              "transcript": "Some people are inspecting the contents of a box.",
              "updated": "2022-12-15 06:13:49.912Z"
            }
          ]
        },
        "id": "ldo275ox71gbbtp",
        "tags": [
          "x895usrl7rwdqq1",
          "btrb7keup275gjf",
          "nb71rws3bp6c7d7"
        ],
        "updated": "2022-12-15 06:17:49.956Z"
      }
    ],
    "tags": [
      {
        "collectionId": "7zoamdyupy1v4pa",
        "collectionName": "tags",
        "created": "2022-12-09 13:43:41.390Z",
        "icon": "exam_kwN9vxvtis.svg",
        "id": "btrb7keup275gjf",
        "isPublish": true,
        "order": 0,
        "updated": "2022-12-19 02:40:06.393Z"
      },
      {
        "collectionId": "7zoamdyupy1v4pa",
        "collectionName": "tags",
        "created": "2022-12-09 13:43:08.082Z",
        "icon": "006_audiobook_nApwBG1W05.svg",
        "id": "x895usrl7rwdqq1",
        "isPublish": true,
        "order": 1,
        "updated": "2022-12-17 02:35:07.147Z"
      },
      {
        "collectionId": "7zoamdyupy1v4pa",
        "collectionName": "tags",
        "created": "2022-11-22 12:02:18.116Z",
        "icon": "exam_O8noHACr3Q.svg",
        "id": "pentyhugxh5ro3e",
        "isPublish": true,
        "order": 0,
        "updated": "2022-12-19 02:43:51.797Z"
      }
    ]
  },
  "id": "ws2q59uc581p7oa",
  "name": "abc",
  "order": 1,
  "questions": [
    "hu8r8kp7qi6lvzw",
    "ilclz85sbv63cx2",
    "rzb43rv6zs1en1g",
    "01eqc9bgwhk4oj8",
    "55q9jjsnbh7ulrm",
    "ldo275ox71gbbtp"
  ],
  "serial": 10,
  "tags": [
    "btrb7keup275gjf",
    "x895usrl7rwdqq1",
    "pentyhugxh5ro3e"
  ],
  "topics": [
    "6ratz4hpf1vocjd",
    "9rft22kzdo1wp8k"
  ],
  "updated": "2022-12-19 04:34:48.293Z"
}
```

**Comment by ganigeorgiev** [comment]
Hm, this could be a regression after the recent expand refactoring (although I remember adding more unit tests, but I guess I've missed a scenario/edge case).

The error is probably a result of `panic` (at first look at the code we are not checking whether the `m.expand` map is initialized).
Could you start the executable with `--debug` and see if there is any stack trace in the console?

Could you also export your collections structure from "Admin UI > Settings > Export collections" and attach the json here?

**Comment by uteminhduc** [comment]
Thanks for your respone
Here is debug log `--debug` after send request
```
[0.00ms] SELECT {{_collections}}.* FROM `_collections` WHERE [[id]] = 'tests' OR LOWER([[name]])='tests' LIMIT 1
[0.00ms] SELECT {{_collections}}.* FROM `_collections` WHERE [[id]] = 'yg2xylk9a3jv7lb' OR LOWER([[name]])='yg2xylk9a3jv7lb' LIMIT 1
[0.00ms] SELECT `tests`.* FROM `tests` WHERE `tests`.`id`='ws2q59uc581p7oa' LIMIT 1
[9.00ms] SELECT {{_collections}}.* FROM `_collections` WHERE [[id]] = 'v6i1h1klj8t22ef' OR LOWER([[name]])='v6i1h1klj8t22ef' LIMIT 1
[0.00ms] SELECT {{_collections}}.* FROM `_collections` WHERE [[id]] = 'v6i1h1klj8t22ef' OR LOWER([[name]])='v6i1h1klj8t22ef' LIMIT 1
[1.00ms] SELECT `questions`.* FROM `questions` WHERE `questions`.`id` IN ('hu8r8kp7qi6lvzw', 'ilclz85sbv63cx2', 'rzb43rv6zs1en1g', '01eqc9bgwhk4oj8', '55q9jjsnbh7ulrm', 'ldo275ox71gbbtp')
[0.00ms] SELECT {{_collections}}.* FROM `_collections` WHERE [[id]] = '63e9guamova874l' OR LOWER([[name]])='63e9guamova874l' LIMIT 1
[0.00ms] SELECT {{_collections}}.* FROM `_collections` WHERE [[id]] = '63e9guamova874l' OR LOWER([[name]])='63e9guamova874l' LIMIT 1
[0.00ms] SELECT `choices`.* FROM `choices` WHERE `choices`.`id` IN ('vsi0uks4iyj9w6w', 'q995onuch75u0w1', '0zhgk79e7cxg0kt', 'zdupl1j9p51xyqv', 'blextbzaae9pr3l', '1en55ygighc1fqp', 'w09a9weli90vyyp', 'zx5j9oc57h6dhil', 'f32vwvsn76ol1yo', '1lfpb5vw9ljqgny', '6oy8m468q3v2prz', '3eog8i0nvok82oi', 'i3e9ed932lxzhdq', '7fb12ca6o91fbir', '6u8oj9y0p74h687', 'fgrd8kuibag2ss2', 'f7eve1wbozruzu6', '8wagn9mbhryp1sa', '3z270afy8bmgvcv', 'lbalsgmn92sigp8', 'o6jonflsq968pi1', 'uo9dh9sxopjy1zi', '0i1smp8ideu62r4', '4sobjk4pxti891h')
[0.00ms] SELECT {{_collections}}.* FROM `_collections` WHERE [[id]] = 'v6i1h1klj8t22ef' OR LOWER([[name]])='v6i1h1klj8t22ef' LIMIT 1
[0.00ms] SELECT {{_collections}}.* FROM `_collections` WHERE [[id]] = 'v6i1h1klj8t22ef' OR LOWER([[name]])='v6i1h1klj8t22ef' LIMIT 1
[0.00ms] SELECT `questions`.* FROM `questions` WHERE `questions`.`id` IN ('hu8r8kp7qi6lvzw', 'ilclz85sbv63cx2', 'rzb43rv6zs1en1g', '01eqc9bgwhk4oj8', '55q9jjsnbh7ulrm', 'ldo275ox71gbbtp')
[0.00ms] SELECT {{_collections}}.* FROM `_collections` WHERE [[id]] = 'v6i1h1klj8t22ef' OR LOWER([[name]])='v6i1h1klj8t22ef' LIMIT 1
[0.00ms] SELECT {{_collections}}.* FROM `_collections` WHERE [[id]] = 'v6i1h1klj8t22ef' OR LOWER([[name]])='v6i1h1klj8t22ef' LIMIT 1
[0.00ms] SELECT `questions`.* FROM `questions` WHERE 0=1
2022/12/24 13:37:31 [PANIC RECOVER] assignment to entry in nil map goroutine 60 [running]:
github.com/labstack/echo/v5/middleware.RecoverConfig.ToMiddleware.func1.1.1()
        /home/runner/go/pkg/mod/github.com/labstack/echo/v5@v5.0.0-20220201181537-ed2888cfa198/middleware/recover.go:72 +0x110
panic({0x1e35180, 0x2aa6490})
        /opt/hostedtoolcache/go/1.19.4/x64/src/runtime/panic.go:884 +0x212
github.com/pocketbase/pocketbase/models.(*Record).MergeExpand(0xc00054ff00, 0xc000170aa0?)
        /home/runner/work/pocketbase/pocketbase/models/record.go:145 +0x357
github.com/pocketbase/pocketbase/daos.(*Dao).expandRecords(0xc000123530?, {0xc000171208, 0x1, 0x18?}, {0xc00015e0d2?, 0x18?}, 0xc00011a570, 0x1)
        /home/runner/work/pocketbase/pocketbase/daos/record_expand.go:219 +0x15e6
github.com/pocketbase/pocketbase/daos.(*Dao).ExpandRecords(0xc000322700?, {0xc000171208, 0x1, 0x1}, {0xc000123530?, 0xc000108800?, 0xc000123470?}, 0x1e59360?)
        /home/runner/work/pocketbase/pocketbase/daos/record_expand.go:40 +0xfc
github.com/pocketbase/pocketbase/apis.EnrichRecords({0x2acb4a8, 0xc000108080}, 0x1e2e9d8b8f8?, {0xc000133208, 0x1, 0x1}, {0x0?, 0x0, 0x0})
        /home/runner/work/pocketbase/pocketbase/apis/record_helpers.go:74 +0x22c
github.com/pocketbase/pocketbase/apis.EnrichRecord(...)
        /home/runner/work/pocketbase/pocketbase/apis/record_helpers.go:54
github.com/pocketbase/pocketbase/apis.(*recordApi).view.func2(0xc00011a558)
        /home/runner/work/pocketbase/pocketbase/apis/record_crud.go:144 +0x7e
github.com/pocketbase/pocketbase/tools/hook.(*Hook[...]).Trigger(0xc0003266c0?, 0x2, {0xc000133320?, 0x1, 0xc000133328})
        /home/runner/work/pocketbase/pocketbase/tools/hook/hook.go:65 +0x1bd
github.com/pocketbase/pocketbase/apis.(*recordApi).view(0xc0001fdc00, {0x2acb4a8, 0xc000108080})
        /home/runner/work/pocketbase/pocketbase/apis/record_crud.go:143 +0x2a5
github.com/pocketbase/pocketbase/apis.LoadCollectionContext.func1.1({0x2acb4a8, 0xc000108080})
        /home/runner/work/pocketbase/pocketbase/apis/middlewares.go:270 +0x152
github.com/pocketbase/pocketbase/apis.ActivityLogger.func1.1({0x2acb4a8, 0xc000108080})
        /home/runner/work/pocketbase/pocketbase/apis/middlewares.go:283 +0x54
github.com/labstack/echo/v5/middleware.CORSConfig.ToMiddleware.func1.1({0x2acb4a8, 0xc000108080})
        /home/runner/go/pkg/mod/github.com/labstack/echo/v5@v5.0.0-20220201181537-ed2888cfa198/middleware/cors.go:143 +0x47e
github.com/pocketbase/pocketbase/apis.LoadAuthContext.func1.1({0x2acb4a8, 0xc000108080})
        /home/runner/work/pocketbase/pocketbase/apis/middlewares.go:216 +0x2a3
github.com/labstack/echo/v5/middleware.SecureConfig.ToMiddleware.func1.1({0x2acb4a8, 0xc000108080})
        /home/runner/go/pkg/mod/github.com/labstack/echo/v5@v5.0.0-20220201181537-ed2888cfa198/middleware/secure.go:142 +0x966
github.com/labstack/echo/v5/middleware.RecoverConfig.ToMiddleware.func1.1({0x2acb4a8, 0xc000108080})
        /home/runner/go/pkg/mod/github.com/labstack/echo/v5@v5.0.0-20220201181537-ed2888cfa198/middleware/recover.go:78 +0x124
github.com/labstack/echo/v5.(*Echo).ServeHTTP.func1({0x2acb4a8, 0xc000108080})
        /home/runner/go/pkg/mod/github.com/labstack/echo/v5@v5.0.0-20220201181537-ed2888cfa198/echo.go:620 +0x117
github.com/labstack/echo/v5/middleware.RemoveTrailingSlashConfig.ToMiddleware.func1.1({0x2acb4a8, 0xc000108080})
        /home/runner/go/pkg/mod/github.com/labstack/echo/v5@v5.0.0-20220201181537-ed2888cfa198/middleware/slash.go:136 +0x211
github.com/labstack/echo/v5.(*Echo).ServeHTTP(0xc000956a00, {0x2ab9e68, 0xc000146000}, 0xc00012a000)
        /home/runner/go/pkg/mod/github.com/labstack/echo/v5@v5.0.0-20220201181537-ed2888cfa198/echo.go:626 +0x234
net/http.serverHandler.ServeHTTP({0xc000122030?}, {0x2ab9e68, 0xc000146000}, 0xc00012a000)
        /opt/hostedtoolcache/go/1.19.4/x64/src/net/http/server.go:2947 +0x30c
net/http.(*conn).serve(0xc000347c20, {0x2aba9a0, 0xc0007b0570})
        /opt/hostedtoolcache/go/1.19.4/x64/src/net/http/server.go:1991 +0x607
created by net/http.(*Server).Serve
        /opt/hostedtoolcache/go/1.19.4/x64/src/net/http/server.go:3102 +0x4db

goroutine 1 [semacquire]:
sync.runtime_Semacquire(0xc00001f9b0?)
        /opt/hostedtoolcache/go/1.19.4/x64/src/runtime/sema.go:62 +0x25
sync.(*WaitGroup)


```
Export collections
[pb_schema.zip](https://github.com/pocketbase/pocketbase/files/10298114/pb_schema.zip)

**Comment by ganigeorgiev** [comment]
@uteminhduc Thank you. The panic from above confirms that it is the nil-map that causing the error.

I've pushed a fix and added a new test in [v0.10.4 release](https://github.com/pocketbase/pocketbase/releases/tag/v0.10.4).

If you are still experiencing issues after the upgrade, please let me know and I'll investigate it in more details.

**Comment by uteminhduc** [comment]
I just tested, the bug was fixed. Thank you @ganigeorgiev .
